pkg/workloads/console/controller.go

```diff
@@ -391,17 +439,28 @@ func (r *reconciler) updateStatus(job *batchv1.Job) error {
 		pod = nil
 	}
 
-	newStatus := calculateStatus(r.console, job, pod)
+	newStatus := calculateStatus(r.console, authorised, job, pod)
 
 	// If there's no changes in status, don't unnecessarily update the object.
 	// This would cause an infinite loop!
 	if reflect.DeepEqual(r.console.Status, newStatus) {
 		return nil
 	}
 
+	if newStatus.Phase == workloadsv1alpha1.ConsolePendingAuthorisation &&
+		r.console.Creating() {
+		auditLog(r.logger, r.console, nil, ConsolePendingAuthorisation, "Console pending authorisation")
+	}
+
+	// Console phase from Pending Authorisation
+	if newStatus.Phase != workloadsv1alpha1.ConsolePendingAuthorisation &&
+		r.console.PendingAuthorisation() {
+		auditLog(r.logger, r.console, nil, ConsoleAuthorised, "Console authorised")
+	}
+
 	// Console phase from Pending to Running
 	if newStatus.Phase == workloadsv1alpha1.ConsoleRunning &&
-		r.console.Status.Phase == workloadsv1alpha1.ConsolePending {
+		r.console.Pending() {
```

Comment:
Lets use the helper functions we have! Plus I think it's easier to read.

Reply:
Nice, agreed!

Reply:
I do think we should have swapped the order of these comparisons though, I find it much easier to visualise the transition when reading `{old,new}` rather than `{new,old}` - but that's out of the scope of this PR!

Reply:
Oh yeah that's totally right. I new something was feeling weird about the comparisons and I think this is it!

Reply:
Ended up just doing this change while I had my editor open.